File: golang/lc-1768.go
package main

func mergeAlternately(word1 string, word2 string) string {
	i, j := 0, 0
	res := make([]byte, 0, len(word1)+len(word2))
	for i < len(word1) && j < len(word2) {
		res = append(res, word1[i], word2[j])
		i++
		j++
	}
	res = append(res, []byte(word1[i:])...)
	res = append(res, []byte(word2[j:])...)
	return string(res)
}

File: js&ts/LeetCode75/lc-1768.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function (word1, word2) {
  let i = 0;
  let j = 0;
  let res = "";
  while (i < word1.length && j < word2.length) {
    res += word1[i++];
    res += word2[j++];
  }
  res = res + word1.substring(i, word1.length);
  res = res + word2.substring(j, word2.length);

  return res;
};
